### DISCUSSION TITLE
有没有什么适合新手参考的, 使用了 MVVM 模式的 .NET 项目推荐?

### DISCUSSION THREAD

**Discussion by Minokori** [BYJRK/DotNet-Discussions#357] [MVVM]
> 本人纯半路出家，入坑.NET 还是研究生导师老是把之前外包写的陈年老代码交给我让我加新功能跟修Bug。但我个人还是挺喜欢写代码的，不是学计算机的结果也就做下去了。有了点小经验就像自己鼓捣鼓捣写点代码，把这些winform的老代码重构或者是做点自己的小程序啥的。

也是看了很多网上关于 MVVM 模式的讨论，但大多数都是给了一些示例代码片段，关于 MVVM 的实际应用的项目实例其实 Github 上也有，但也是见仁见智，每个人的实现跟理解都不太一样。(其实我觉得之前的帖子里说的 #11 
"如果把UI换掉了，VM如果不需要改就可以说是符合MVVM"说的就挺好的，教我这种小白一抓一准)微软自己的 MVVM Sample 也说的不是很清楚，有没有什么让人一看就觉得，“哇，MVVM就该是这样” 的项目，能够帮助理解 MVVM 正确，或者说合适的项目组织方式？

~~*Talk is cheap. Show me the code.*~~

**Comment by CodingOctocat** [comment]
不嫌弃的话，可以看看我的一个小项目：https://github.com/CodingOctocat/IPConfig ，不是完美 MVVM，因为代价太大不值得，但尽可能平衡地使用 MVVM 了。没啥注释，组织结构、命名什么的我尽可能做到规范，应该还是能看的😂。

**Reply by dongpo1996** [reply]
看了 还是挺规范的

**Comment by showerror** [comment]
我最近在看官方的WPF Gallery，Win11 UI风格，代码逻辑也非常严谨，项目地址https://github.com/lepoco/wpfui

**Reply by BYJRK** [reply]
官方的目前可以在 [.NET 9 预览版](https://github.com/dotnet/core/blob/main/release-notes/9.0/preview/preview4/wpf.md)中使用。不过目前还是更推荐 [UI.WPF.Modern](https://github.com/iNKORE-NET/UI.WPF.Modern) 一些

**Comment by sealoyal2018** [comment]
试试这个 chatgpt 桌面版

![预览图](https://github.com/sealoyal2018/ChatBox/blob/master/images/common.gif)

项目地址：[传送门](https://github.com/sealoyal2018/ChatBox.git)
